smartcontactmanager/src/main/java/com/scm/smartcontactmanager/service/category/CategoryServiceImpl.java

```diff
@@ -1,8 +1,137 @@
+
+	@Autowired
+	private CategoryDao categoryDao;
+	@Autowired
+	private ContactCategoryRelationRepository contactCategoryRelationRepository ;
+
+	@Override
+	public CategoryResponseDto createCategory(@Valid CategoryRequestDto categoryRequestDto) {
+		Category category = mapToEntity(categoryRequestDto);
+		Category findCategory = categoryDao.findCategory(categoryRequestDto.getName(),categoryRequestDto.getType());
+		if(findCategory!=null) {
+			throw new ServiceException(HttpStatus.valueOf(409), "Category already exsists");
+		}else {
+			Category createCategory = categoryDao.createCategory(category);
+			
+			CategoryResponseDto savedResponse = mapToResponseDto(createCategory);
+			return savedResponse;
+		}
+		
+	}
+
+	Category mapToEntity(CategoryRequestDto categoryRequestDto) {
+		Category category = new Category();
+		category.setName(categoryRequestDto.getName());
+		category.setType(categoryRequestDto.getType());
+		category.setUserId(categoryRequestDto.getUserId());
+		category.setByAdmin(false);
+		category.setCreatedOn(new Date());
+		category.setUpdatedOn(new Date());
+		return category;
+	}
+
+	CategoryResponseDto mapToResponseDto(Category category) {
+		CategoryResponseDto categoryResponseDto = new CategoryResponseDto();
+		categoryResponseDto.setName(categoryResponseDto.getName());
+		categoryResponseDto.setType(categoryResponseDto.getType());
+		categoryResponseDto.setUserId(categoryResponseDto.getUserId());
+		categoryResponseDto.setByAdmin(false);
+		return categoryResponseDto;
+	}
+	
+	
+	@Override
+	public List<Category> getListOfCategory(){
+		List<Category> categoryList = categoryDao.getListOfCategory();
+		if(categoryList==null) {
+			throw new ServiceException(HttpStatusCode.valueOf(404), "Categories not found");
+		}
+		return categoryList;
+	}
 
+	@Override
+	public void addContactsToCategory(int id, ContactIds contactIds) throws Exception {
+		Category category = categoryDao.getCategoryById(id).get();
+		List<Contact> contactList = categoryDao.getContactdByIds(contactIds.getContactId());
+		if(category != null && contactList.size() == contactIds.getContactId().size()) {
+			List<ContactCategoryRelation> relations = new ArrayList<>();
+			for(int cId: contactIds.getContactId()) {
+				ContactCategoryRelation relation = ContactCategoryRelation.builder()
+						                                       .categoryId(id)
+						                                       .contactId(cId)
+						                                       .build();
+		
+				relations.add(relation);
+			}
+			List<ContactCategoryRelation> concatrelation = categoryDao.addContactsToCategory(relations);
+		} else {
+			throw new Exception("Invalid Input!");
+		}
+		
+	}
+    @Override
+	public  List<Contact> getContactsByCategoryId(int id) {
+    	
+    	Optional<Category> categoryById = categoryDao.getCategoryById(id);	
+    	if(categoryById.isEmpty()) {
+    		throw new ServiceException(HttpStatusCode.valueOf(404),"Category not found with id: "+id);
```

Comment:
Error message should be "Category with id {} not found"

Reply:
Addressed